budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -252,14 +533,175 @@ def test_url_path(self):
+        url_path = result["url_path"]
+        response = self.client.get(url_path)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, province)
 
-class ProvInfraProjectContentTestCase(APITestCase):
-    fixtures = ["test-prov-infra-project-content"]
 
-    def test_project_detail_content(self):
-        project = ProvInfraProject.objects.first()
-        url = project.get_absolute_url()
+class ProvInfraProjectSnapshotTestCase(APITransactionTestCase):
+    def setUp(self):
+        self.project = ProvInfraProject.objects.create(IRM_project_id=1)
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.quarter_1 = Quarter.objects.create(number=1)
+        self.irm_snapshot_1 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=self.quarter_1,
+            date_taken=date(year=2019, month=1, day=1),
+        )
+        self.project_snapshot_1 = ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot_1,
+            project=self.project,
+            status="Construction",
+        )
+
+        self.quarter_2 = Quarter.objects.create(number=2)
+        self.irm_snapshot_2 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=self.quarter_2,
+            date_taken=date(year=2019, month=6, day=1),
+        )
+        self.project_snapshot_2 = ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot_2, project=self.project, status="Completed"
+        )
+
+    def test_latest_status_in_the_content(self):
+        url = self.project.get_absolute_url()
         response = self.client.get(url)
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertContains(response, project.name)
+        self.assertContains(response, '"status": "Completed"')
+        self.assertNotContains(response, '"status": "Construction"')
+
+    def test_latest_in_the_same_year(self):
+        latest = ProvInfraProjectSnapshot.objects.filter(project=self.project).latest()
+
+        self.assertEqual(self.project_snapshot_2, latest)
+
+
+class ProvInfraProjectSnapshotDifferentYearsTestCase(APITransactionTestCase):
+    def setUp(self):
+        self.project = ProvInfraProject.objects.create(IRM_project_id=1)
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.quarter_1 = Quarter.objects.create(number=1)
+
+        self.irm_snapshot_1 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=self.quarter_1,
+            date_taken=date(year=2019, month=1, day=1),
+        )
+        self.project_snapshot_1 = ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot_1, project=self.project,
+        )
+
+        self.irm_snapshot_2 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
```

Comment:
`financial_year` should be different - `date_taken` isn't relevant to this test so let's make it identical to demonstrate that.

Reply:
Solved.